### PR TITLE
Add pod DeleteImmediate function

### PR DESCRIPTION
**Pull request by mcornea**

### CODE REVIEW

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -223,6 +223,31 @@ func (builder *Builder) DeleteAndWait(timeout time.Duration) (*Builder, error) {
 	return builder, nil
 }
 
+// DeleteImmediate removes the pod immediately object and resets the builder object.
```
Comment (achuzhoy) [inline pkg/pod/pod.go:226]:
```suggestion
// DeleteImmediate removes the pod immediately and resets the builder object.
```

File: pkg/pod/pod.go [hunk 2]
```diff
@@ -223,6 +223,31 @@ func (builder *Builder) DeleteAndWait(timeout time.Duration) (*Builder, error) {
...
+func (builder *Builder) DeleteImmediate() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Immediately deleting pod %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("pod cannot be deleted because it does not exist")
```
Comment (trewest) [inline pkg/pod/pod.go:236]: We just return nil if it doesn't exist instead of an error

Comment (trewest) [reply pkg/pod/pod.go:236]: The way the pod Delete method is currently written is outdated (you don't have to change it)

File: pkg/pod/pod.go [hunk 3]
```diff
@@ -223,6 +223,31 @@ func (builder *Builder) DeleteAndWait(timeout time.Duration) (*Builder, error) {
...
+	}
+
+	err := builder.apiClient.Pods(builder.Definition.Namespace).Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{GracePeriodSeconds: ptr.To(int64(0))})
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete pod: %w", err)
```
[the remark this answers is not in the document]
Comment (mcornea) [reply pkg/pod/pod.go]: tnx, updated.

File: pkg/pod/pod.go (hunk 2 above)
Comment (mcornea) [reply pkg/pod/pod.go]: tnx, updated.

**Review by achuzhoy — APPROVED** [review]

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**